### PR TITLE
ToJSON: allow custom as_json function; JobsetInputs, Jobsets: update schema to align with the API

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

File: src/lib/Hydra/Component/ToJSON.pm [hunk 1]
```diff
@@ -10,6 +10,10 @@ use JSON;
 sub TO_JSON {
     my $self = shift;
 
+    if ($self->can("as_json")) {
+      return $self->as_json();
```
Comment (grahamc) [inline src/lib/Hydra/Component/ToJSON.pm:14]:
```suggestion
        return $self->as_json();
```

File: src/lib/Hydra/Controller/Jobset.pm [hunk 2]
```diff
@@ -270,8 +270,8 @@ sub updateJobset {
     $jobset->jobsetinputs->delete;
 
     if ($type == 0) {
-        foreach my $name (keys %{$c->stash->{params}->{jobsetinputs}}) {
-            my $inputData = $c->stash->{params}->{jobsetinputs}->{$name};
+        foreach my $name (keys %{$c->stash->{params}->{inputs}}) {
+            my $inputData = $c->stash->{params}->{inputs}->{$name};
```
Comment (grahamc) [inline src/lib/Hydra/Controller/Jobset.pm:274]: This JS function will need to be updated to account for this change: https://github.com/NixOS/hydra/blob/master/src/root/edit-jobset.tt#L223-L243

**Comment by grahamc** [comment]
Let's look at and maybe merge https://github.com/NixOS/hydra/pull/942 first, to catch API test breaks.

**Comment by grahamc** [comment]
Hrm. This is strange, the tests passed in GitHub Actions and `flake check` passed too, but on deploy it failed...

```
make[2]: Entering directory '/build/source/t'
( STDERR )  job  1    {UNKNOWN}: Bareword "JSON::true" not allowed while "strict subs" in use at /build/source/src/lib/Hydra/Schema/JobsetInputs.pm line 142.
( STDERR )  job  1    Bareword "JSON::false" not allowed while "strict subs" in use at /build/source/src/lib/Hydra/Schema/JobsetInputs.pm line 142.
( STDERR )  job  1    Compilation failed in require at /nix/store/qm2vwb1h2206hswgnf80m9mv1q5fzg4m-hydra-perl-deps/lib/perl5/site_perl/5.32.0/Class/C3/Componentised.pm line 146. at /nix/store/qm2vwb1h2206hswgnf80m9mv1q5fzg4m-hydra-perl-deps/lib/perl5/site_perl/5.32.0/Class/C3/Componentised.pm line 151
( STDERR )  job  1    Compilation failed in require at /build/source/src/script/hydra-init line 4.
( STDERR )  job  1    BEGIN failed--compilation aborted at /build/source/src/script/hydra-init line 4.
( STDERR )  job  1    Died at /build/source/t/lib/Setup.pm line 76.
```

gonna need to figure that out, I don't understand.

**Comment by cole-h** [comment]
For posterity's sake: the above issue was caused by missing `use JSON;`s (https://github.com/NixOS/hydra/pull/945).

**Pull request merged**